apps/wizard/utils/io.py

```diff
@@ -137,3 +145,27 @@ def get_new_grapher_datasets_and_their_previous_versions(session: Session) -> Di
             new_datasets[ds_id] = previous_dataset
 
     return new_datasets
+
+
+def get_all_changed_catalog_paths(files_changed: Dict[str, Dict[str, str]]) -> List[str]:
+    """Get all changed steps and their downstream dependencies."""
+    dataset_catalog_paths = []
+
+    # Get catalog paths of all datasets with changed files.
+    for step_path in get_changed_steps(files_changed):
```

Comment:
This would not catch snapshots. To do that, you could do, e.g. `if step_path.startswith("snapshots/")` or something like that.

Reply:
Good catch! I made it work for snapshots, too.